src/parser.rs

```diff
@@ -124,3 +126,136 @@ impl Statement {
+
+    mod get {
+        use super::*;
+
+        #[test]
+        fn returns_fail_on_get_without_key() {
+            let statement = get_statement!("GET");
+            assert_eq!(statement, Statement {
+                stype: StatementType::Fail,
+                key: None,
+                value: None
+            });
+        }
+
+        #[test]
+        fn returns_get() {
+            let statement = get_statement!("GET MY_KEY");
+            assert_eq!(statement, Statement {
+                stype: StatementType::Get,
+                key: Some("MY_KEY".to_owned()),
+                value: None
+            });
+        }
+
+        #[test]
+        fn returns_get_ignoring_next_keys() {
+            let statement = get_statement!("GET KEY1 KEY2 KEY3");
+            assert_eq!(statement, Statement {
+                stype: StatementType::Get,
+                key: Some("KEY1".to_owned()),
+                value: None
+            });
+        }
+    }
+
+    mod del {
+        use super::*;
+
+        #[test]
+        fn returns_fail_on_del_without_key() {
+            let statement = get_statement!("DEL");
+            assert_eq!(statement, Statement {
+                stype: StatementType::Fail,
+                key: None,
+                value: None
+            });
+        }
+
+        #[test]
+        fn returns_del() {
+            let statement = get_statement!("DEL MY_KEY");
+            assert_eq!(statement, Statement {
+                stype: StatementType::Del,
+                key: Some("MY_KEY".to_owned()),
+                value: None
+            });
+        }
+
+        #[test]
+        fn returns_del_ignoring_next_keys() {
+            let statement = get_statement!("DEL KEY1 KEY2 KEY3");
+            assert_eq!(statement, Statement {
+                stype: StatementType::Del,
+                key: Some("KEY1".to_owned()),
+                value: None
+            });
+        }
+    }
+
+    mod set {
+        use super::*;
+
+        #[test]
+        fn returns_fail_on_set_without_key() {
+            let statement = get_statement!("SET");
+            assert_eq!(statement, Statement {
+                stype: StatementType::Fail,
+                key: None,
+                value: None
+            });
+        }
+
+        #[test]
+        fn returns_fail_on_set_without_value() {
+            let statement = get_statement!("SET MY_KEY");
+            assert_eq!(statement, Statement {
+                stype: StatementType::Fail,
+                key: None,
+                value: None
+            });
+        }
+
+        #[test]
+        fn returns_set() {
+            let statement = get_statement!("SET MY_KEY MY_VALUE");
+            assert_eq!(statement, Statement {
+                stype: StatementType::Set,
+                key: Some("MY_KEY".to_owned()),
+                value: Some("MY_VALUE".to_owned())
+            });
+        }
+
+        #[test]
+        fn returns_set_considering_all_next_values() {
```

Comment:
This might not be the best desired behaviour, we might consider adding a regex to allow for setting multiple key-value pairs with a single SET command. With values like strings being quotation limited, e.g:
```
SET key1 value key2 "hi hello world"
```

Reply:
I don't know which is your case. maybe we can see what [`redis`](https://redis.io/) does in this case. Anyway I threat a change as a new PR